dashboard/lib/pd/payment/payment_calculator_csf.rb

```diff
@@ -22,20 +22,12 @@ def qualified?(total_teacher_attendance_days)
 
     # Gets session attendance summaries for a workshop.
     # CSF workshops have different attendance rules. Workshops are always exactly one day (session),
-    # and every teacher in the section counts but only those who have solved >= 10 puzzles are qualified (see #teacher_qualified?)
+    # and every attending teacher counts but only those who have solved >= 10 puzzles are qualified (see #teacher_qualified?)
     # @param workshop [Pd::Workshop]
     # @return [Array<SessionAttendanceSummary>] summary of attendance for each session.
     def get_session_attendance_summaries(workshop)
-      # Anyone in the section (even if it's been deleted) with an enrollment counts as attended for CSF
-      teacher_ids = Follower.with_deleted.where(section_id: workshop.section_id).pluck(:student_user_id)
-
-      workshop_enrollments = workshop.enrollments.all
-      enrollment_ids = teacher_ids.map do |teacher_id|
-        workshop_enrollments.find {|enrollment| enrollment.user_id == teacher_id}
-      end.compact.map(&:id)
-
-      # Return exactly one session (day), with the raw hours from the session.
-      [SessionAttendanceSummary.new(workshop.sessions.first.hours, enrollment_ids)]
+      # Return exactly one session (day)
+      [super.first]
```

Comment:
CSF workshops are always supposed to be a single session. We don't stop organizers from creating multiple sessions, but we will only recognize and pay for one. 

Note, that logic is preserver from before (see line 37 in the red).